File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Events/ObjectContainerEventArgs.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using System;
using Db4objects.Db4o;

namespace Db4objects.Db4o.Events
{
	/// <summary>Arguments for container related events.</summary>
	/// <remarks>Arguments for container related events.</remarks>
	/// <seealso cref="IEventRegistry">IEventRegistry</seealso>
	public class ObjectContainerEventArgs : EventArgs
	{
		private readonly IObjectContainer _container;

		public ObjectContainerEventArgs(IObjectContainer container)
		{
			_container = container;
		}

		public virtual IObjectContainer ObjectContainer
		{
			get
			{
				return _container;
			}
		}
	}
}

File: db4oj/core/src/com/db4o/events/ObjectContainerEventArgs.java
/* Copyright (C) 2004 - 2006 Versant Inc. http://www.db4o.com */

package com.db4o.events;

import com.db4o.*;

/**
 * Arguments for container related events.
 * 
 * @see EventRegistry
 */
public class ObjectContainerEventArgs extends EventArgs {

	private final ObjectContainer _container;

	public ObjectContainerEventArgs(ObjectContainer container) {
		_container = container;
	}
	
	/**
	 * @sharpen.property
	 */
	public ObjectContainer objectContainer() {
		return _container;
	}
}
